### ISSUE TITLE
500 on repositories after git version 2.37.1.windows.1

### ISSUE THREAD

**Issue by ChameleonRed** [go-gitea/gitea#20426]
### Description

git version 2.37.1.windows.1

```
2022/07/20 17:16:25 ...dules/git/command.go:146:RunWithContext() [D] C:\Users\Cezary Wagner\gitea-repositories\cezary.wagner\medptr-v2.git: C:\Program Files\Git\cmd\git.exe -c credential.helper= -c protocol.version=2 -c uploadpack.allowfilter=true -c uploadpack.allowAnySHA1InWant=true -c filter.lfs.required= -c filter.lfs.smudge= -c filter.lfs.clean= rev-parse
2022/07/20 17:16:25 ...ules/context/repo.go:598:RepoAssignment() [E] RepoAssignment Invalid repo C:\Users\Cezary Wagner\gitea-repositories\cezary.wagner\medptr-v2.git: exit status 128 - fatal: detected dubious ownership in repository at 'C:/Users/Cezary Wagner/gitea-repositories/cezary.wagner/medptr-v2.git'
	To add an exception for this directory, call:
	
		git config --global --add safe.directory 'C:/Users/Cezary Wagner/gitea-repositories/cezary.wagner/medptr-v2.git'
	
	Set the environment variable GIT_TEST_DEBUG_UNSAFE_DIRECTORIES=true and run
	again for more information.
	
2022/07/20 17:16:25 ...s/context/context.go:214:HTML() [D] Template: status/500
```

After setting GIT_TEST_DEBUG_UNSAFE_DIRECTORIES log is more detailed

```
2022/07/20 17:17:14 ...dules/git/command.go:146:RunWithContext() [D] C:\Users\Cezary Wagner\gitea-repositories\cezary.wagner\medptr-v2.git: C:\Program Files\Git\cmd\git.exe -c credential.helper= -c protocol.version=2 -c uploadpack.allowfilter=true -c uploadpack.allowAnySHA1InWant=true -c filter.lfs.required= -c filter.lfs.smudge= -c filter.lfs.clean= rev-parse
2022/07/20 17:17:14 ...ules/context/repo.go:598:RepoAssignment() [E] RepoAssignment Invalid repo C:\Users\Cezary Wagner\gitea-repositories\cezary.wagner\medptr-v2.git: exit status 128 - warning: 'C:/Users/Cezary Wagner/gitea-repositories/cezary.wagner/medptr-v2.git' is owned by:
		'S-1-5-21-3618141223-1333197053-3853304746-1001'
	but the current user is:
		'S-1-5-18'
	fatal: detected dubious ownership in repository at 'C:/Users/Cezary Wagner/gitea-repositories/cezary.wagner/medptr-v2.git'
	To add an exception for this directory, call:
	
		git config --global --add safe.directory 'C:/Users/Cezary Wagner/gitea-repositories/cezary.wagner/medptr-v2.git'
	
2022/07/20 17:17:14 ...s/context/context.go:214:HTML() [D] Template: status/500
```

### Gitea Version

1.16.9

### Can you reproduce the bug on the Gitea demo site?

No

### Git Version

git version 2.37.1.windows.1

### Operating System

Window 10

### How are you running Gitea?

Windows Service

**Comment by silverwind** [comment]
`safe.directory` issues should be fixed in 1.17 or higher, try to upgrade.

**Comment by silverwind** [comment]
The fix was in https://github.com/go-gitea/gitea/pull/19870 which is only in 1.17, so yes, workaround or upgrade please.
